pkg/controllers/controlplanemachineset/cluster_operator_test.go

```diff
@@ -128,29 +139,226 @@ var _ = Describe("Cluster Operator Status with a running controller", func() {
+			logger = test.NewTestLogger()
+
+			// Create a control plane machine set.
+			cpms = resourcebuilder.ControlPlaneMachineSet().WithName(clusterControlPlaneMachineSetName).WithNamespace(reconciler.Namespace).Build()
+			Expect(k8sClient.Create(ctx, cpms)).To(Succeed())
+
+			conds := []metav1.Condition{
+				newCondition(conditionAvailable, metav1.ConditionTrue, reasonAllReplicasAvailable, ""),
+				newCondition(conditionProgressing, metav1.ConditionFalse, reasonAllReplicasUpdated, ""),
+				newCondition(conditionDegraded, metav1.ConditionFalse, reasonAsExpected, ""),
+			}
+
+			// Setting its status to available.
+			Eventually(komega.UpdateStatus(cpms, func() {
+				cpms.Status = resourcebuilder.ControlPlaneMachineSetStatusBuilder{}.WithConditions(conds).Build()
+			})).Should(Succeed())
+
+			// Waiting for cluster operator to become available too.
+			Eventually(func() (bool, error) {
+				err := k8sClient.Get(context.Background(), client.ObjectKey{Name: operatorName}, co)
+				if err != nil {
+					return false, fmt.Errorf("failed to get Cluster Operator %v, %w", operatorName, err)
+				}
+
+				return co != nil && co.Status.Conditions != nil, nil
+			}).Should(BeTrue())
+		})
+
+		It("Do not update the status if conditions stay the same", func() {
+			// Explicitly call updateClusterOperatorStatus with the same conditions to make sure it doesn't send a request
+			conds := []metav1.Condition{
+				newCondition(conditionAvailable, metav1.ConditionTrue, reasonAllReplicasAvailable, ""),
+				newCondition(conditionProgressing, metav1.ConditionFalse, reasonAllReplicasUpdated, ""),
+				newCondition(conditionDegraded, metav1.ConditionFalse, reasonAsExpected, ""),
+			}
+
+			cpms.Status.Conditions = conds
+			Expect(reconciler.updateClusterOperatorStatus(context.Background(), logger.Logger(), cpms)).To(Succeed())
+
+			Expect(logger.Entries()).To(BeEmpty())
+		})
+
+		It("Update the status if conditions changed", func() {
+			// Explicitly call updateClusterOperatorStatus with different conditions to make sure it sends a request
+			conds := []metav1.Condition{
+				newCondition(conditionAvailable, metav1.ConditionFalse, reasonUnavailableReplicas, "Missing 3 available replica(s)"),
+				newCondition(conditionProgressing, metav1.ConditionFalse, reasonOperatorDegraded, ""),
+				newCondition(conditionDegraded, metav1.ConditionTrue, reasonUnmanagedNodes, "Found 3 unmanaged node(s)"),
+			}
+
+			cpms.Status.Conditions = conds
+			Expect(reconciler.updateClusterOperatorStatus(context.Background(), logger.Logger(), cpms)).To(Succeed())
+
+			Expect(logger.Entries()).To(ConsistOf(
+				test.LogEntry{
+					Level:   4,
+					Message: "Syncing cluster operator status",
+					KeysAndValues: []interface{}{
+						"available",
+						string(metav1.ConditionFalse),
+						"progressing",
+						string(metav1.ConditionFalse),
+						"degraded",
+						string(metav1.ConditionTrue),
+						"upgradable",
+						string(metav1.ConditionFalse),
```

Comment:
Nit, this would be easier to read/understand as key value pairs on the same line
```suggestion
						"available", string(metav1.ConditionFalse),
						"progressing", string(metav1.ConditionFalse),
						"degraded", string(metav1.ConditionTrue),
						"upgradable", string(metav1.ConditionFalse),
```